### DISCUSSION TITLE
Image ID

### DISCUSSION THREAD

**Discussion by peeeyow** [epwalsh/obsidian.nvim#355] [Q&A] [answered]
Hello,
Does anyone know how to control the parsing and saving of image names? I'm looking for something similar to the note_id_func, but applied by the ObsidianPasteImg command. I'm planning to automatically prefix the names with a timestamp to avoid collisions in the future. Thanks.

**Comment by epwalsh** [comment, marked as answer]
Hey @peeeyow, as of ae39ba3 this is now possible by adding an `image_name_func` to your config. Here's an example from my personal dotfiles: 

https://github.com/epwalsh/dotfiles/blob/e6a89e358b2bdf3247b47d62bd4de05a98550be1/neovim/lua/plugins/notes.lua#L113-L123

**Reply by peeeyow** [reply]
Thanks for this @epwalsh. Much appreciated.
